docs/DeveloperGuide.md

```diff
@@ -1,38 +1,312 @@
 # Developer Guide
 
+* [Acknowledgements](#acknowledgements)
+* [Setting up, getting started](#setting-up-getting-started)
+* [Design](#design)
+    * [Architecture](#architecture)
+    * [UI Component](#ui-component)
+    * [Parser Component](#parser-component)
+    * [Recipe Component](#recipe-component)
+    * [RecipeList Component](#recipelist-component)
+    * [WeeklyPlan Component](#weeklyplan-component)
+    * [Database Component](#database-component)
+* [Implementation](#implementation)
```

Comment:
Perhaps you could add a numbering format in this table of content and the subsequent subheaders so that it is easier to view, for eg. 
```
- 1. Acknowledgements
- 2. Setting up, getting started
- 3. Design 
    - 3.1 Architecture 
    - 3.2 UI Component
```